112.md

```diff
@@ -0,0 +1,257 @@
+This effectively hides metadata from external users. See drawbacks below.
+
+It reserves 8 event kinds (1059, 400-406) for immediate use.
+
+- [`1059`](https://github.com/nostr-protocol/nips/pull/468) - encrypted gift-wrap, can be used by many protocols
+- `400`  - wrapped kind: create encrypted channel
+- `401`  - wrapped kind: invite to encrypted channel
+- `402`  - wrapped kind: change channel metadata
+- `403`  - wrapped kind: send encrypted chat message
+- `404`  - wrapped kind: moved to new channel
+- `405`  - wrapped kind: delegate new owner
+- `406`  - wrapped kind: accept owner delegation
+
+
+## Kind 1059: Encrypted Gift Wrap
+
+For all encrypted group events, the client first generates the approrpiate kind 4XX message, as below, and signs it with the user's identity key.   
+
+Then they generate a new public/private keypair, and uses this to sign new kind 1059 message.
+
+The `content` of that message is a [NIP-44](https://github.com/nostr-protocol/nips/pull/574) encrypted JSON string.
+
+The content is encrypted using the public key of the destination (the chat room public key, or the invitee public key.
+
+All events called "wrapped-kind XX" refer to these kind 1059 gift wrapped events.
+
+For more information on gift-wrapping, see: [NIP-59](https://github.com/nostr-protocol/nips/pull/468)
+
+## Wrapped Kind 400: Create Encrypted channel
+
+Create a Encrypted event channel.
+
+In the channel creation `content` field, Client SHOULD include basic channel metadata (`name`, `about` and `picture`).   
+
+This is akin to NIP-28, kind 40 but using gift-wrap encryption.
+
+On creating a channel, the creator MUST also generate a unique/new private-public key pair which will serve as the `shared-secret` for a given channel. 
+
+This is the "destination pubkey" for the channel creation.   
+
+A list of recommended relays SHOULD be added to the channel-create message
+
+```json
+{
+    "content": "{\"name\": \"Demo Channel\", \"about\": \"A test channel.\", \"picture\": \"https://placekitten.com/200/200\"}",
+    "tags": [["r", "wss://relay-01"], ["r", "wss://relay-02"], ...]]
+}
+```
+
+The wrapped INNER event id for the channel creation should be saved.   This is the event id that will be used in all future INNER references to the channel creation.
+
+Channel creation events MUST be sent to the "destination pubkey" of the channel itself, not to any specific user.
+
+As a rule, this "destination pubkey" of the channel MUST used as the sole identifier and external filter that clients use to retrieve events for the channel.
+
+There are no other public tags or metadata, all other kinds and tags are "inner" (inside the wrapped message).  
+
+## Wrapped Kind 401: Invitation to Encrypted Channel
+ 
+Invitations are sent directly to the participants of the channel.
+
+```json
+{
+    "content": "optional personalized invitation message",
+    "tags": [["e", "channel-create-event-id", "relay-url"], ["privkey", "channel-private-key"]]
```

Comment:
I am missing what is 'privkey' mentioned here in kind 401? There is 'channel-private-key', which is the shared secret private key used commonly by all to encrypt/decrypt channel messages. But what is 'privkey'?

Reply:
That's just a literal tag string, `"privkey"`. Square bracket notation should probably be used for non-literals.

Reply:
Ah yes. I missed that, very basic thing. Am back looking at json after many weeks!..